server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -334,4 +405,70 @@ private void deleteStaleCommits() {
             logger.info("Exception while deleting stale commits from remote segment store, will retry delete post next commit", e);
         }
     }
+
+    /**
+     * Updates the last refresh time and refresh seq no which is seen by local store.
+     */
+    private void updateLocalRefreshTimeAndSeqNo() {
+        RemoteRefreshSegmentTracker segmentTracker = pressureService.getRemoteRefreshSegmentTracker(indexShard.shardId());
+        segmentTracker.updateLocalRefreshTimeMs(pendingRefreshTimeMs);
+        segmentTracker.updateLocalRefreshSeqNo(segmentTracker.getLocalRefreshSeqNo() + 1);
+    }
+
+    /**
+     * Updates the last refresh time and refresh seq no which is seen by remote store.
+     */
+    private void updateRemoteRefreshTimeAndSeqNo(RemoteRefreshSegmentTracker segmentTracker, long refreshTimeMs, long refreshSeqNo) {
+        segmentTracker.updateRemoteRefreshTimeMs(refreshTimeMs);
+        segmentTracker.updateRemoteRefreshSeqNo(refreshSeqNo);
+    }
+
+    /**
+     * Returns map of file name to size of the input segment files. Tries to reuse existing information by caching the size
+     * data, otherwise uses {@code storeDirectory.fileLength(file)} to get the size. This method also removes from the map
+     * such files that are not present in the list of segment files given in the input.
+     *
+     * @param segmentFiles list of segment files for which size needs to be known
+     * @return the map as mentioned above
+     */
+    private Map<String, Long> createSizeMap(Collection<String> segmentFiles) {
+        // Create a map of file name to size
+        Map<String, Long> sizeMap = segmentFiles.stream()
+            .filter(file -> !EXCLUDE_FILES.contains(file))
+            .collect(Collectors.toMap(Function.identity(), file -> {
+                if (latestFileNameSizeOnLocalMap.containsKey(file)) {
+                    return latestFileNameSizeOnLocalMap.get(file);
+                }
+                long fileSize = 0;
+                try {
+                    fileSize = storeDirectory.fileLength(file);
+                    latestFileNameSizeOnLocalMap.put(file, fileSize);
+                } catch (IOException e) {
+                    logger.warn(new ParameterizedMessage("Exception while reading the fileLength of file={}", file), e);
+                }
+                return fileSize;
+            }));
+        // Remove keys from the fileSizeMap that do not exist in the latest segment files
+        latestFileNameSizeOnLocalMap.entrySet().removeIf(entry -> sizeMap.containsKey(entry.getKey()) == false);
+        return sizeMap;
```

Comment:
Do we need to create and return the new map each time or we can re-use `latestFileNameSizeOnLocalMap`?

Reply:
it's reusing the `latestFileNameSizeOnLocalMap` only.